xapian-core/weight/bm25plusweight.cc

```diff
@@ -110,7 +109,7 @@ BM25PlusWeight::get_sumpart(Xapian::termcount wdf, Xapian::termcount len,
     // Parameter delta (δ) is a pseudo tf value to control the scale of the
     // tf lower bound. δ can be tuned for e.g from 0.0 to 1.5 but BM25+ can
     // still work effectively across collections with a fixed δ = 1.0
-    RETURN((termweight * (wdf_double / denom)) + param_delta);
+    RETURN((termweight * ((((param_k1 + 1) * wdf_double) / denom) + param_delta)));
 }
```

Comment:
There's now a `((` and `))` around the whole return expression, which just makes it harder to read the expression - this means exactly the same:

```
RETURN(termweight * ((((param_k1 + 1) * wdf_double) / denom) + param_delta));
```

But also `(x * y) / z` and `x * y / z` mean the same since `*` and `/` have the same precedence (http://en.cppreference.com/w/cpp/language/operator_precedence) and `+` has lower precedence, so you could actually lose another two pair of parentheses:

```
RETURN(termweight * ((param_k1 + 1) * wdf_double / denom + param_delta));
```

This step's more arguable - some people prefer explicit parentheses, especially when there are operators with different precedence involved.  But let's definitely lose the extra outer pair (and similarly below).

Also in a case like this where the initial implementation has a bug, it's a really good idea to add a regression testcase which will fail if the fix ever got undone.  Bugs have a tendency to recur - if the code got misimplemented in a particular way, it's natural that someone reworking it may make the same mistake again, and it's also good not to repeat the same mistake as it looks bad to users, management, etc - and adding regression testcases is a big help in preventing that.